### PR TITLE
Add grad accumulation control

### PR DESCRIPTION
**Pull request by KSGulin**
Add control over gradient accumulation steps and nominal batch size via the arg `--gradient_accum_steps`. If the arg is not set, the default nominal batch size of 64 is honored. If it is set, the nominal batch size is set to be the product of the the batch size and the gradient accumulation steps.

Tests:
`sparseml.yolov5.train --epoch 1`
`sparseml.yolov5.train --epoch 1 --gradient-accum-steps 4 --batch-size 16`

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -8,6 +8,7 @@ Pillow>=7.1.2
 PyYAML>=5.3.1
 requests>=2.23.0
 scipy>=1.4.1
+torchvision>=0.8.1
```
Comment (corey-nm) [inline requirements.txt:11]: Is this change related? Otherwise lgtm

Comment (KSGulin) [reply requirements.txt]: Whoops that's unrelated. Thanks for catching

**Review by corey-nm — APPROVED** [review]

**Review by rahul-tuli — APPROVED** [review]
